File: tests/benchmark/vulnerable/go/sqli_001.go
// BENCHMARK: vulnerable - sqli
package main

import (
	"database/sql"
	"fmt"
	"net/http"

	_ "github.com/lib/pq"
)

var db *sql.DB

func searchHandler(w http.ResponseWriter, r *http.Request) {
	name := r.URL.Query().Get("name")
	rows, err := db.Query(fmt.Sprintf("SELECT * FROM products WHERE name='%s'", name))
	if err != nil {
		http.Error(w, err.Error(), 500)
		return
	}
	defer rows.Close()
	fmt.Fprintf(w, "found rows: %v", rows)
}

func main() {
	http.HandleFunc("/search", searchHandler)
	http.ListenAndServe(":8080", nil)
}

File: tests/benchmark/vulnerable/javascript/sqli_001.js
// BENCHMARK: vulnerable - sqli
const express = require('express');
const { Pool } = require('pg');
const app = express();
const pool = new Pool();

app.use(express.json());

app.post('/login', async (req, res) => {
    const { username } = req.body;
    const result = await pool.query(`SELECT * FROM users WHERE username = '${username}'`);
    res.json(result.rows);
});

app.listen(3000);
